File: Vjudge/A_Closest_Point.cpp
#include <bits/stdc++.h>
using namespace std;

void solve()
{
    int n;
    cin >> n;
    vector<int> a(n);
    for (int i = 0; i < n; i++)
        cin >> a[i];
    if (n > 2)
    {
        cout << "NO" << '\n';
        return;
    }

    if (a[1] == a[0] + 1)
        cout << "NO" << '\n';
    else
        cout << "YES" << '\n';
}

int main()
{

    int t;
    cin >> t;
    while (t--)
    {
        solve();
    }
}

File: Vjudge/A_Closest_Point.java
import java.util.*;

public class A_Closest_Point {

    static void solve() {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int[] a = new int[n];
        for (int i = 0; i < n; i++) {
            a[i] = sc.nextInt();
        }

        if (n > 2) {
            System.out.println("NO");
            return;
        }

        if (a[1] == a[0] + 1) {
            System.out.println("NO");
        } else {
            System.out.println("YES");
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();
        while (t-- > 0) {
            solve();
        }
    }
}
